orc8r/cloud/go/swagger/combine.go

```
@@ -0,0 +1,222 @@
+/*
+ Copyright 2020 The Magma Authors.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree.
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package swagger
+
+import (
+	"github.com/golang/glog"
+
+	"regexp"
+	"sort"
+
+	"magma/orc8r/cloud/go/tools/swaggergen/generate"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+// Load specs from YAML format to Swagger structs.
+func Load(common string, configs []string) ([]generate.SwaggerSpec, generate.SwaggerSpec, error) {
+	specs, err := loadSwaggerSpecs(configs)
+	if err != nil {
+		return nil, generate.SwaggerSpec{}, err
+	}
+
+	commonSpec, err := loadCommonSpec(common)
+	if err != nil {
+		return nil, generate.SwaggerSpec{}, err
+	}
+
+	return specs, commonSpec, nil
+}
+
+/*
+CombineStr is a wrapper function around the function Combine()
+which serves to combine multiple swagger specs whose contents
+are passed in. The output is the merged copy of the swagger spec
+*/
+func CombineStr(common string, configs []string) (string, error) {
+	cfgs, commonCfg, err := Load(common, configs)
+	if err != nil {
+		return "Error: Load of Swagger Specs Failed", err
+	}
+	
+	combinedSpec, warnings := Combine(commonCfg, cfgs)
+	glog.V(2).Infof("Combining Swagger Specs Together \n")
+	merrs, _ := warnings.(*multierror.Error)
```

- There's no need to cast the error here. We can just log it as is, if it's not nil
- There's a design decision to be made here -- should we *fail* when there's a merge error, or should we just log? I think we should swallow the error and log, since things *can* still function correctly even if there are merge errors
- That said, it should be an `Error` log rather than `Warning`, because it is indeed an error. Use warning sparingly -- basically, only when you have a situation that is *probably* an error, but not certain

---

I concur that we should swallow the error and log since the merge errors just indicate overwrites and that doesn't really affect much.